### PR TITLE
T-17778 Add v1 Endpoints that use new DB driver

### PR DESCRIPTION
**Pull request by commoddity**
This PR does the following:
- Updates the `main.go` file to include both the `V1 Cache` and `V1 Router`, thus enabling the V1 endpoints.

- Adds the E2E test file for testing V1 endpoints, updates the Makefile with new test targets and updates the .github workflows to run the V1 tests in the CI.

**Merge this PR only after both https://github.com/pokt-foundation/pocket-http-db/pull/94 and https://github.com/pokt-foundation/pocket-http-db/pull/95**

### CODE REVIEW

**Review by ricantar — APPROVED** [review]
Sweet

File: main.go [hunk 1]
```diff
@@ -7,10 +7,14 @@ import (
...
-	"github.com/pokt-foundation/pocket-http-db/router"
-	postgresdriver "github.com/pokt-foundation/portal-api-go/postgres-driver"
+	v0router "github.com/pokt-foundation/pocket-http-db/router"
+	v1router "github.com/pokt-foundation/pocket-http-db/v1_router"
+	v0postgresdriver "github.com/pokt-foundation/portal-api-go/postgres-driver"
+	v1postgresdriver "github.com/pokt-foundation/portal-db/postgres-driver"
 	"github.com/pokt-foundation/utils-go/environment"
 	"github.com/sirupsen/logrus"
+
+	_ "net/http/pprof"
```
Comment (adshmh) [inline main.go:17]: Is this needed?

File: main.go [hunk 2]
```diff
@@ -39,7 +43,20 @@ func gatherOptions() options {
 	}
 }
 
-func cacheHandler(router *router.Router, cacheRefresh int64, log *logrus.Logger) {
+/* DEPRECATED -- v0 cache handler using portal-api-go driver */
+func v0CacheHandler(router *v0router.Router, cacheRefresh int64, log *logrus.Logger) {
+	for {
+		time.Sleep(time.Duration(cacheRefresh) * time.Minute)
```
Comment (adshmh) [inline main.go:49]: time.Sleep, should be avoided whenever possible, especially when the expected sleep time is in minutes.
[time.After](https://pkg.go.dev/time#After) is commonly used for this scenario.

Comment (adshmh) [reply main.go:49]: In addition, it seems something that is an internal details of the `router` performed by an external function.
An internal method on the router itself (with a better name, e.g. `CacheRefresh()` run in its own go routine seems a better choice.

File: main.go [hunk 3]
```diff
@@ -86,8 +119,13 @@ func main() {
 
 	wg.Add(1)
 
-	go httpHandler(router, options.port, log)
-	go cacheHandler(router, options.cacheRefresh, log)
+	go httpHandler(v0Router, router, options.port, log)
+
+	/* DEPRECATED -- cache handler using portal-api-go driver */
+	go v0CacheHandler(v0Router, options.cacheRefresh, log)
```
Comment (adshmh) [inline main.go:125]: Please see above review comment regarding this method and the corresponding one for v1.

File: main_e2e_v1_test.go [hunk 4]
```diff
@@ -0,0 +1,1509 @@
...
+	"github.com/pokt-foundation/portal-db/types"
+	"github.com/stretchr/testify/suite"
+)
+
+var (
+	testCtx       = context.Background()
+	mockTimestamp = time.Date(2022, 11, 11, 11, 11, 11, 0, time.UTC)
+)
+
+type DBClientTestSuite struct {
```
Comment (adshmh) [inline main_e2e_v1_test.go:20]: Not a blocker, but this does not really need to be an exported type.

File: main_e2e_v1_test.go [hunk 5]
```diff
@@ -0,0 +1,1509 @@
...
+							{
+								ID:     "test_app_47hfnths73j2se",
+								UserID: "test_user_1dbffbdfeeb225",
+								Name:   "pokt_app_123",
+								URL:    "https://test.app123.io",
+								Dummy:  true,
+								Status: types.InService,
+								GatewayAAT: types.GatewayAAT{
+									Address:              "test_34715cae753e67c75fbb340442e7de8e",
+									ApplicationPublicKey: "test_11b8d394ca331d7c7a71ca1896d630f6",
```
Comment (adshmh) [inline main_e2e_v1_test.go:463]: Not a blocker, but moving all this data to json-formatted files with reasonable names under testdata directory will make this code more readable.

File: main_e2e_v1_test.go [hunk 6]
```diff
@@ -0,0 +1,1509 @@
...
+				ts.Equal(test.application.Limit, application.Limit)
+				ts.Equal(test.application.NotificationSettings, application.NotificationSettings)
+				ts.NotEmpty(application.CreatedAt)
+				ts.NotEmpty(application.UpdatedAt)
+
+				application, err = ts.client2.GetApplicationByID(testCtx, createdApp.ID)
+				ts.Equal(test.err, err)
+				ts.Equal(createdApp.ID, application.ID)
+				ts.Equal(test.application.Dummy, application.Dummy)
+				ts.Equal(test.application.Status, application.Status)
```
Comment (adshmh) [inline main_e2e_v1_test.go:1083]: Is it possible to use a library for this, e.g. https://github.com/google/go-cmp, as we do with other repositories, to reduce the large number of comparisons?

**Review by adshmh — CHANGES_REQUESTED** [review]
Thank you for the update. This is a great improvement over the previous PR: most of the added code is in a single file and is for e2e testing. Just a few suggestions (if followed, could significantly reduce the amount of actual code added here)

**Review by ezeike — APPROVED** [review]
These PRs are much clearer. Looks good to me.

File: main.go (hunk 1 above)
Comment (commoddity) [reply main.go:17]: Nope, that's for the memory profiler. Good catch; I will remove.

File: main.go (hunk 2 above)
Comment (commoddity) [reply main.go:49]: Sounds good. I suggest we do this in a separate PR  as this is not related to the v1 change (this is how the cache refresh has worked since we first created PHD).

File: main.go (hunk 3 above)
Comment (commoddity) [reply main.go:125]: This is regarding the `time.Sleep` call?

File: main_e2e_v1_test.go (hunk 4 above)
Comment (commoddity) [reply main_e2e_v1_test.go:20]: Agreed; I'll update to be unexported.

File: main_e2e_v1_test.go (hunk 5 above)
Comment (commoddity) [reply main_e2e_v1_test.go:463]: Agreed; I actually have a suggestion for reducing duplication of test data for the PHD test docker container but I suggest we discuss that in another context as it can be an eventual improvement after this PR is merged. WDYT?

File: main_e2e_v1_test.go (hunk 6 above)
Comment (commoddity) [reply main_e2e_v1_test.go:1083]: The only issue with this is that the timestamps (`CreatedAt`, `UpdatedAt`) don't match and are not deterministic since the record is created by the test; otherwise I would just compare the entire structs. Open to a more readable approach for sure but again I suggest we do this in a subsequent PR.

File: main.go (hunk 2 above)
Comment (commoddity) [reply main.go:47]: @pablocampogo WDYT?

File: main.go (hunk 3 above)
Comment (adshmh) [reply main.go:123]: No, it is regarding the this item: ```In addition, it seems something that is an internal details of the router performed by an external function.
An internal method on the router itself (with a better name, e.g. CacheRefresh() run in its own go routine seems a better choice.```

Comment (commoddity) [reply main.go:123]: Ah gotcha. Yeah I agree that implementation can be improved but still suggest we don't do it in this PR.

File: main_e2e_v1_test.go (hunk 6 above)
Comment (adshmh) [reply main_e2e_v1_test.go:1083]: Sure, handling this on a seaparate PR is OK.

File: main.go (hunk 2 above)
Comment (commoddity) [reply main.go:47]: Added to Height, resolving this conversation.

File: cache/listener.go [hunk 7]
```diff
@@ -236,6 +236,9 @@ func (c *Cache) listen() {
 
 	for {
 		n := <-c.reader.NotificationChannel()
-		go c.parseNotification(*n)
+		// nil check is needed because the v1 DB alerts will send a nil notification for user_access table
```
Comment (adshmh) [inline cache/listener.go:239]: Not a blocker: this needs a unit test to cover the case the driver returns a nil notification.

**Review by adshmh — APPROVED** [review]
LGTM

File: cache/listener.go (hunk 7 above)
Comment (commoddity) [reply cache/listener.go:239]: Noted, will add afterwards.
